src/main/java/ch/njol/skript/sections/SecLoop.java

```diff
@@ -101,8 +103,10 @@ protected TriggerItem walk(Event e) {
 		if (iter == null || !iter.hasNext()) {
 			exit(e);
 			debug(e, false);
+			loopCounter = 0;
```

Comment:
This won't work when the loop is run twice at the same time (e.g. because of a recursive function)